x-pack/plugins/apm/server/utils/merge_periods_timeseries.test.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Coordinate } from '../../typings/timeseries';
+import { mergePeriodsTimeseries } from './merge_periods_timeseries';
+
+describe('mergePeriodsTimeseries', () => {
+  it('returns empty array when timeseries is empty', () => {
+    expect(
+      mergePeriodsTimeseries({
+        currentPeriodTimeseries: [],
+        previousPeriodTimeseries: [],
+      })
+    ).toEqual([]);
+  });
+  it('returns empty array when comparison timeseries is undefined', () => {
+    expect(
+      mergePeriodsTimeseries({
+        currentPeriodTimeseries: [],
+        previousPeriodTimeseries: undefined,
+      })
+    ).toEqual([]);
+  });
+
+  it('returns empty array when current period timeseries is empty', () => {
```

Comment:
Thanks for adding this! IMHO this is a bug. At least, I'm not sure why this function would not return data for the previous period if there is data. Instead of merging buckets, have you considered offsetting the data for the previous period by the time difference between `end` and `comparisonEnd` (or `start` and `comparisonStart`, not sure what the implications are)?

Reply:
@dgieselaar I changed the merge function to calculate the x-axis from the previous timeseries based on the time difference (currentPeriodStart and previousPeriodStart)

https://github.com/elastic/kibana/pull/90128/files#diff-0b2c86f706e4bcb3ab0cfaaf6cc7a6cee303d02b59c9b11a6af847eb5916aedbR11